### DISCUSSION TITLE
Dipping my toes into local LLM - recommended container image for ollama?

### DISCUSSION THREAD

**Discussion by psyciknz** [icereed/paperless-gpt#39] [General]
I've been following this project for a few weeks and wanted to try it out now.

If there a recommended container for ollama?  I was intending to run it on a 2nd machine and more than likely behind a reverse proxy, but initially just want a container that paperless-gpt can utilise for it's processing.

**Comment by icereed** [comment]
Hey 👋 
Welcome aboard 🫶

On with operating system do you plan to run ollama? Sounds like Linux to me, but I don’t want to merely assume.

**Reply by psyciknz** [reply]
I have two proxmox machines.

I did just try running ollama with lama2 on a secondary machine.  Has plenty of ram but no gpu.  

It was running as a container on an lxc machine as I tend to run docker that way.

After giving it one document and leaving it for about 30 mins. I see it's cpu bound.  And suspect I might need to love it to my primary server where I can use the gpu. Only an nvidia 1030 bit the machine has more cpu as well.

But I thought in 30 mins it might had done something. 

Oh so is can be anything. Linux primarily.

**Reply by icereed** [reply]
I tried LLMs once without GPU. It’s almost unusable 😅
LLMs need GPUs. That’s why chip makers are getting so rich with GPUs right now 😂

If you want to tune your setup even more and hide ollama behind an API key you can try https://docs.litellm.ai/docs/simple_proxy. 

It adds a little complexity to your setup, but you can plug a lot of different LLMs into it.

**Reply by psyciknz** [reply]
So that would be proxying the public llms. Like OpenAI etc?

**Reply by icereed** [reply]
It supports local LLMs, too 😊

**Reply by icereed** [reply]
https://docs.litellm.ai/docs/providers/ollama#using-ollama-apichat

**Reply by icereed** [reply]
I’m not sure yet, if LiteLLM supports the endpoints for Visual LLMs like discussed here #20
